java/rocksjni/transaction.cc

```diff
@@ -1582,7 +1556,7 @@ jbyte Java_org_rocksdb_Transaction_getState(JNIEnv* /*env*/, jobject /*jobj*/,
   }
 
   assert(false);
-  return 0xFF;
+  return static_cast<char>(-1);
```

Comment:
Why the change from `0xFF` to `-1`?

Reply:
Msvc complains with constant overflow warning (signed char overflow) . This is only solution I have found to fix that warning.